### PR TITLE
Feature/4015 componentize eula screen

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-4015

#### Changes proposed in this Pull Request:

- Componentize AcceptEula component
- Added docs

#### Screenshots / Screen Recording (if applicable)
- 
Screenshot 2023-04-19 at 3 49 52 PM

#### To Test:

- yarn start:example

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/docs/AcceptEula.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| onEulaChanged            | Function to call when the state of the checkbox is changed | `function`                                                                                               | yes      |                                                   |
+| loadEula                 | Function to call to retrieve the eulaContent               | `function`                                                                                               | yes      |                                                   |
+| htmlEula                 | Prop to render EULA as HTML                                | `boolean`                                                                                                | no       |                                                   |
+| eulaError                | Error message if the EULA fails to load                    | `string`                                                                                                 | no       |                                                   |
```
Comment (huayunh) [inline login-workflow/docs/AcceptEula.md:33]: This prop name is unclear.  Is `eulaError` a text? A callback function triggered whenever there is an error? To pass in a JavaScript `Error` object? Users are expected to mostly  guess what a prop does based on just the way it is named.

Comment (huayunh) [reply login-workflow/docs/AcceptEula.md:33]: There are a lot of similar issues in the current API table. I did not highlight all of them; I think you can infer what kind of changes I am looking for based on my other suggestions. I recommend you applying design thinking principles you learned a while ago and doing a user testing activity with Manoj. You provide only the prop name, and let him guess what the prop is controlling.

File: login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        slots = {},
+        slotProps = {},
+    } = props;
+
+    const defaultClasses = useUtilityClasses(props);
+    const TermsAndConditions = slots.termsAndConditions ?? FormControlLabel;
+    const eulaContentInternals = eulaContent ?? eulaError ?? loaderText;
+
+    useEffect(() => {
+        void loadEula();
```
[the remark this answers is not in the document]
Comment (huayunh) [reply login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx:118]: I am also expecting the component to handle more logic for the user. I expect some kind of `useState` to set the EULA area to be either "loading", "loaded" or "error", and based on this internal state, the subscreen renders the appropriate content. Right now, it looks like the adopters are the ones being forced to maintain this state. For me, this is not a great adopter user experience.

File: login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                                onEulaChanged(event.target.checked)
+                            }
+                        />
+                    )
+                }
+                label={agreeTerms}
+                sx={{ flex: '0 0 auto', mr: 0, mt: 2, ...termsAndConditionsStyles }}
+                className={cx(defaultClasses.termsAndConditions, classes.termsAndConditions)}
+                {...TermsAndConditionsProps}
+                {...slotProps.termsAndConditions}
```
Comment (huayunh) [inline login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx:162]: Let's use only the `slotProps` here and get rid of that `TermsAndConditionsProps`.

File: login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                </Typography>
+            )}
+            {htmlEula && (
+                <Box
+                    sx={{ flex: '1 1 0px', overflow: 'auto', ...eulaContentStyles }}
+                    className={cx(defaultClasses.eulaContent, classes.eulaContent)}
+                    component={slots.eulaContent}
+                    dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(eulaContentInternals) }}
+                    {...EulaContentProps}
+                    {...slotProps.eulaContent}
```
Comment (huayunh) [inline login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx:142]: Use only slotProps

File: login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+    return (
+        <>
+            {!htmlEula && (
+                <Typography
+                    sx={{ flex: '1 1 0px', overflow: 'auto', ...loaderStyles }}
+                    className={cx(defaultClasses.loaderText, classes.loaderText)}
+                    component={slots.loaderText}
+                    {...LoaderTextProps}
+                    {...slotProps.loaderText}
```
Comment (huayunh) [inline login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx:130]: use only slotProps

**Review by huayunh — CHANGES_REQUESTED** [review]

File: login-workflow/docs/AcceptEula.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+| agreeTerms               | Prop to override terms and conditions text                 | `string`                                                                                                 | no       | "I have read and agree to the Terms & Conditions" | 
+| eulaControl              | Prop to override the terms and conditions checkbox         | `ReactElement`                                                                                           | no       |                                                   | 
+| termsAndConditionsStyles | Prop to override terms and conditions styles               | `SxProps<Theme>`                                                                                         | no       |                                                   |
```
[the remark this answers is not in the document]
Comment (surajeaton) [reply login-workflow/docs/AcceptEula.md:36]: termsAndConditions is FormControlLabel which replaces to the whole component including label and checkbox.

File: login-workflow/docs/AcceptEula.md (hunk 1 above)
Comment (surajeaton) [reply login-workflow/docs/AcceptEula.md:33]: > This prop name is unclear. Is `eulaError` a text? A callback function triggered whenever there is an error? To pass in a JavaScript `Error` object? Users are expected to mostly guess what a prop does based on just the way it is named.

Yes eulaError is string set to transitErrorMessage

**Comment by surajeaton** [comment]
@huayunh Updated the PR. Please review

**Review by huayunh — APPROVED** [review]
The change looks good

**Pull request merged**